File: TweetApp/FrontEnd/tweet-app/src/app/models/Tweet.ts
import { User } from "./User";

export interface Tweet {
	id: number;
	subject: string;
	tag?: string;
	userId: number;
	user: User|null;
	datePosted: string;
}

export interface CreateTweet {
	id: number;
	subject: string;
	tag?: string;
}

File: TweetApp/Backend/TweetApp.Repository/Entities/Tweet.cs
#nullable disable
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TweetApp.Repository.Entities
{
    public class Tweet
    {
        [Key]
        public int Id { get; set; }

        [MaxLength(50)]
        public string Tag { get; set; }

        [MaxLength(144)]
        public string Subject { get; set; }

        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; }
        public DateTime DatePosted { get; set; } = DateTime.UtcNow;
    }
}
